### ISSUE TITLE
[WIP] [Improvements]: General improvements / Bugs

### ISSUE THREAD

**Issue by jurrejansen** [Shelf-nu/shelf.nu#708]
### Contact Details

jurre@shelf.nu

### Is your feature request related to a problem? Please describe?

A bunch of visual and quality of life improvements.

## Bookings

When configuring Start and End time/date I find it annoying that you have to click a date and put a time before the field is filled. Would like to suggest that when you click the Start date/time and End date/time it instantly fills a date and time. 

https://github.com/Shelf-nu/shelf.nu/assets/76112272/4e6bb62d-20d8-4d8f-93b9-5e14e2c67d51

Of course how I use the feature when testing is probably not very similar how it is being used in the wild but still. I can imagine some organisations will use it like this:

- Custodian shows up without notice and tell equipment manager what he/she wants
- Equipment manager creates a booking and instantly performs Check-out
- Equipment manager gives custodian the assets

I think in these cases the suggestions below will make things more user friendly.

- [ ] When you click the Start date/time field it should auto fill the current date. Additionally the time should auto fill with the current time +10 minutes. The date picker modal should stay open so the user can change the date and time to their liking.
- [ ] When you click the End date/time field it should auto fill the current date + 1 day. Additionally the time should auto fill with the current time +10 minutes. The date picker modal should stay open so the user can change the date and time to their liking.

## Dropdown menu when clicking account element

- [x] Icons do not have proper container and stroke width
- [x] Account element is not 100% width when username and email are short

https://github.com/Shelf-nu/shelf.nu/assets/76112272/f6d4d053-eff6-47a4-85a9-139bfbd35de3

https://github.com/Shelf-nu/shelf.nu/assets/76112272/dac10029-5bc0-4bbc-a935-0c2be3a384e2

```
<svg width=" 100%" height=" 100%" viewBox="0 0 16 16" fill="none" xmlns="http://www.w3.org/2000/svg">
<path d="M13.3334 14C13.3334 13.0696 13.3334 12.6044 13.2186 12.2259C12.9601 11.3736 12.2931 10.7067 11.4408 10.4482C11.0623 10.3333 10.5971 10.3333 9.66675 10.3333H6.33342C5.40304 10.3333 4.93785 10.3333 4.55932 10.4482C3.70705 10.7067 3.04011 11.3736 2.78157 12.2259C2.66675 12.6044 2.66675 13.0696 2.66675 14M11.0001 5C11.0001 6.65685 9.65694 8 8.00008 8C6.34323 8 5.00008 6.65685 5.00008 5C5.00008 3.34315 6.34323 2 8.00008 2C9.65694 2 11.0001 3.34315 11.0001 5Z" stroke="currentColor" stroke-width="1.5" stroke-linecap="round" stroke-linejoin="round"/>
</svg>
```

```
<svg width=" 100%" height=" 100%" viewBox="0 0 16 16" fill="none" xmlns="http://www.w3.org/2000/svg">
<path d="M4.66667 5.66667H8M4.66667 8H10M6.45583 12H10.8C11.9201 12 12.4802 12 12.908 11.782C13.2843 11.5903 13.5903 11.2843 13.782 10.908C14 10.4802 14 9.9201 14 8.8V5.2C14 4.07989 14 3.51984 13.782 3.09202C13.5903 2.71569 13.2843 2.40973 12.908 2.21799C12.4802 2 11.9201 2 10.8 2H5.2C4.0799 2 3.51984 2 3.09202 2.21799C2.71569 2.40973 2.40973 2.71569 2.21799 3.09202C2 3.51984 2 4.0799 2 5.2V13.557C2 13.9122 2 14.0898 2.07282 14.1811C2.13615 14.2604 2.23218 14.3066 2.33369 14.3065C2.45042 14.3063 2.58911 14.1954 2.8665 13.9735L4.45681 12.7012C4.78168 12.4413 4.94411 12.3114 5.12499 12.219C5.28547 12.137 5.45628 12.0771 5.63281 12.0408C5.83178 12 6.0398 12 6.45583 12Z" stroke="currentColor" stroke-width="1.5" stroke-linecap="round" stroke-linejoin="round"/>
</svg>
```

```
<svg width=" 100%" height=" 100%" viewBox="0 0 16 16" fill="none" xmlns="http://www.w3.org/2000/svg">
<path d="M10.6667 11.3333L14 8M14 8L10.6667 4.66667M14 8H6M6 2H5.2C4.0799 2 3.51984 2 3.09202 2.21799C2.7157 2.40973 2.40973 2.71569 2.21799 3.09202C2 3.51984 2 4.07989 2 5.2V10.8C2 11.9201 2 12.4802 2.21799 12.908C2.40973 13.2843 2.71569 13.5903 3.09202 13.782C3.51984 14 4.0799 14 5.2 14H6" stroke="currentColor" stroke-width="1.5" stroke-linecap="round" stroke-linejoin="round"/>
</svg>
```

## Breadcrumbs

- [ ] Remove the 'Home' icon in the breadcrumbs element
![Screenshot 2024-02-01 at 16 49 48](https://github.com/Shelf-nu/shelf.nu/assets/76112272/3204036d-6f1a-4c41-a711-8ce1ffde8bc6)

- [x] Can we cut off a current page in breadcrumbs when name is very long?
![Screenshot 2024-02-20 at 11 58 59](https://github.com/Shelf-nu/shelf.nu/assets/76112272/388bac04-8bc4-491d-9a4f-67ea4f7e9784)
![Screenshot 2024-02-20 at 12 02 44](https://github.com/Shelf-nu/shelf.nu/assets/76112272/88a054fc-c95c-4177-849c-c9628d04835b)

## Front-end improvements on asset page

- [x] Spacing between elements (you can check proper spacing here: https://www.figma.com/file/ykvc0MhPeNLsrJ4tJsKosQ/Shelf-%7C-UX-Scenarios?type=design&node-id=3437-54808&mode=design)
https://github.com/Shelf-nu/shelf.nu/assets/76112272/4d4e9b26-907d-482a-8d8e-7c1d923b5a5a

- [x] Asset image + description should be 1 element (also on Mobile)
https://github.com/Shelf-nu/shelf.nu/assets/76112272/d1044d04-4a74-48b1-a882-208eb43700ae
![Screenshot 2024-02-05 at 16 56 00](https://github.com/Shelf-nu/shelf.nu/assets/76112272/44e4caaf-7fac-4c83-a33f-4dcb0fd67265)

- [x] Remove location badge that is located next to status badge, showing it only once located in the element that shows other fields is enough
![Screenshot 2024-03-01 at 10 06 29](https://github.com/Shelf-nu/shelf.nu/assets/76112272/86ca62dc-d775-4f2e-921d-4ca2b8bf1c48)

- [x] The height of dropdown field in modal that appears when Assigning custody  is not correct
![Screenshot 2024-03-01 at 11 05 22](https://github.com/Shelf-nu/shelf.nu/assets/76112272/c05269bd-8ca3-4428-9e91-1b94510adfcb)

- [x] The height of dropdown field in modal that appears when Creating a booking is not correct
![Screenshot 2024-03-01 at 11 27 20](https://github.com/Shelf-nu/shelf.nu/assets/76112272/e7538840-b17b-4541-b53e-3e2c7a791094)
![Screenshot 2024-03-01 at 11 54 11](https://github.com/Shelf-nu/shelf.nu/assets/76112272/5fb7d1e1-1eac-4e95-a645-34ddb2fcb7a7)

## Improvements related to scanning

- [x] Only possible to use the in-app scanner once, an after you have to refresh browser. Navigating to another page doesn't work and you have use the browser's refresh button

https://github.com/Shelf-nu/shelf.nu/assets/76112272/312e6d09-c51d-4781-8329-4e21923e0074

- [x] After scanning a code and ending up on an asset page we used to have the Action dropdown automatically pop up. This doesn't happen anymore

https://github.com/Shelf-nu/shelf.nu/assets/76112272/9ef68df5-a6c0-4462-b8e9-4d6c78bdd35f

## Pagination

- [x] Make the dropdown used for selecting amount of rows in table same height as the other pagination elements
![Screenshot 2024-03-01 at 10 51 58](https://github.com/Shelf-nu/shelf.nu/assets/76112272/9c6acfa9-032a-4994-983a-ab7d04403521)
![Screenshot 2024-03-01 at 10 53 42](https://github.com/Shelf-nu/shelf.nu/assets/76112272/633eecaf-8734-4497-89ab-5a9489b41c0b)

### Dashboard

- [x] Location Ratio -> Location ratio
- [x] Newest Assets -> Newest assets

### Assets

- [x] New Asset -> New asset

### Asset

- [x] Give custody -> Assign custody + same change in modal heading
![Screenshot 2024-03-01 at 11 03 15](https://github.com/Shelf-nu/shelf.nu/assets/76112272/ccc5deac-5a2b-462f-b29c-7ddaceecb1cd)

- [x] Update Location -> Update location + same change in modal heading

### Categories

- [x] New Category -> New category
- [x] Search by category name -> Search by category name
![Screenshot 2024-03-01 at 11 16 27](https://github.com/Shelf-nu/shelf.nu/assets/76112272/d678aa4b-0738-4aa0-b187-6d24f956443b)

### Tags

- [x] Search by tag name -> Search by tag

### Locations

- [x] Add Location -> New location
- [x] Search by location name -> search by location

### Location

Manage Assets -> Manage assets
Search by asset name -> Search by asset

### Bookings

- [x] Search by booking name -> Search by booking

**Comment by jurrejansen** [comment]
General

- [ ] In most views there is a unnecessary border located on the right.
![Screenshot 2024-02-20 at 12 01 08](https://github.com/Shelf-nu/shelf.nu/assets/76112272/a52ec0a1-c244-4273-b634-05f019898d12)




Bookings

- [ ] Make entire row clickable when adding / removing assets in the Manage assets modal, instead of having to click precisely on the checkbox

**Comment by DonKoko** [comment]
> Of course how I use the feature when testing is probably not very similar how it is being used in the wild but still. I can imagine some organisations will use it like this:
> 
>     Custodian shows up without notice and tell equipment manager what he/she wants
>     Equipment manager creates a booking and instantly performs Check-out
>     Equipment manager gives custodian the assets
> 
> I think in these cases the suggestions below will make things more user friendly.
> 
> When you click the Start date/time field it should auto fill the current date. Additionally the time should auto fill with the current time +10 minutes. The date picker modal should stay open so the user can change the date and time to their liking.
> When you click the End date/time field it should auto fill the current date + 1 day. Additionally the time should auto fill with the current time +10 minutes. The date picker modal should stay open so the user can change the date and time to their liking.

@jurrejansen This is something we solved in a bit different way. I think what you are suggesting can be done but its a bit tricky and I am not sure how much better it is than what we currently have.
If you look at the current version, what do you think could be improved on that.

**Comment by DonKoko** [comment]
>  Remove the 'Home' icon in the breadcrumbs element

This is something we discussed in detail with carlos and decided to keep it. I tried removing it but it felt kinda confusing why does it say there the same thing as it does in the title. WE just need some way to symbolize its breadcrumbs. We could change the icon tho, if we find something better.

**Comment by DonKoko** [comment]
>  can we make the pagination element sticky above the Modal footer

Kinda complex because it affects lots of views and places. That would require to be handled separately

> Date picker of custom field (Can we use the date picker that we have on bookings? it works much better considering that some of our users want to add items that they purchased years ago - the current UX is not allowing for that with ease)

this is doable but not so easy. Dealing with dates is always hard because of formats and so on so it requires quite some testing and will block the whole PR. Can you create a separate issue for improvement for htis one

**Comment by carlosvirreira** [comment]
> > Date picker of custom field (Can we use the date picker that we have on bookings? it works much better considering that some of our users want to add items that they purchased years ago - the current UX is not allowing for that with ease)
> 
> this is doable but not so easy. Dealing with dates is always hard because of formats and so on so it requires quite some testing and will block the whole PR. Can you create a separate issue for improvement for htis one

No problem - can make a separate low priority issue for this.

**Comment by jurrejansen** [comment]
> @jurrejansen This is something we solved in a bit different way. I think what you are suggesting can be done but its a bit tricky and I am not sure how much better it is than what we currently have. If you look at the current version, what do you think could be improved on that.

@DonKoko Indeed, reported before default Start and End times were set to 08:00 and 18:00. Fields actually totally unfilled which made it annoying. Way better now.

What we could do perhaps:

- [ ] If current time is before 08:00, Start date/time = current day 08:00 | End date/time = current day 18:00
- [ ] If current time is 08:00 - 16:59, Start date/time = current day current time + 10m | End date/time = current day 18:00
- [ ] If current current time is 17:00 - 23:59, Start date/time = current day current time + 10m | End date/time = next day 08:00

Or maybe a simpler variation of this. My goal would be that if a user wants to check out equipment instantly, they do not have to touch the Start date/time field at all.

@carlosvirreira

**Comment by carlosvirreira** [comment]
> > @jurrejansen This is something we solved in a bit different way. I think what you are suggesting can be done but its a bit tricky and I am not sure how much better it is than what we currently have. If you look at the current version, what do you think could be improved on that.
> 
> @DonKoko Indeed, reported before default Start and End times were set to 08:00 and 18:00. Fields actually totally unfilled which made it annoying. Way better now.
> 
> What we could do perhaps:
> 
> * [ ]  If current time is before 08:00, Start date/time = current day 08:00 | End date/time = current day 18:00
> * [ ]  If current time is 08:00 - 16:59, Start date/time = current day current time + 10m | End date/time = current day 18:00
> * [ ]  If current current time is 17:00 - 23:59, Start date/time = current day current time + 10m | End date/time = next day 08:00
> 
> Or maybe a simpler variation of this. My goal would be that if a user wants to check out equipment instantly, they do not have to touch the Start date/time field at all.
> 
> @carlosvirreira

I think your suggestions are nice @jurrejansen - however, if the goal is to have a user check out the equipment instantly (start date / now) then the return date needs to be filled. 

My worry here is that we might be adding complexity and taking time off from building other areas of the app if this ends up being quite complex to manage.

**Comment by jurrejansen** [comment]
> > Remove the 'Home' icon in the breadcrumbs element
> 
> This is something we discussed in detail with carlos and decided to keep it. I tried removing it but it felt kinda confusing why does it say there the same thing as it does in the title. WE just need some way to symbolize its breadcrumbs. We could change the icon tho, if we find something better.

@DonKoko @carlosvirreira It's basically a quick way to get back to Asset index (for mobile mainly as desktop always has the menu item on the left) implying that is the Home of the app. Guess that's true in a way. Not sure.
 
About the showing the same thing twice (title and breadcrumb). In the more updated designs I moved away from using a header when it is not needed, especially on index views where it's just taking up extra space (see Shelf | Library in Figma). The breadcrumbs are a good enough indicator of where you are as a user imo.

Anyways, I personally feel we could lose it and save screen real estate in both these places.

https://github.com/Shelf-nu/shelf.nu/assets/76112272/ea850968-859a-47eb-8fe3-46b583bcf022
